Upstream patch, already in master but not in 9.2.6
Tweaked slightly for applying on 9.2.6 (conan patch is strict)

commit a3d59cded884364a53d3b333735e967e52844fbf
Author: Ben Boeckel <ben.boeckel@kitware.com>
Date:   Mon Jun 27 19:49:32 2022 -0400

    vtkCGNSReader: remove dead code

diff --git a/IO/CGNS/vtkCGNSReader.cxx b/IO/CGNS/vtkCGNSReader.cxx
index f9883b17bf..80e83035d0 100644
--- a/IO/CGNS/vtkCGNSReader.cxx
+++ b/IO/CGNS/vtkCGNSReader.cxx
@@ -58,7 +58,6 @@
 #include <algorithm>
 #include <cmath>
 #include <cstdlib>
-#include <functional>
 #include <iterator>
 #include <map>
 #include <set>
@@ -4221,17 +4220,6 @@ int vtkCGNSReader::GetUnstructuredZone(
   return 0;
 }
 
-//----------------------------------------------------------------------------
-class WithinTolerance : public std::binary_function<double, double, bool>
-{
-public:
-  result_type operator()(first_argument_type a, second_argument_type b) const
-  {
-    bool result = (std::fabs(a - b) <= (a * 1E-6));
-    return (result_type)result;
-  }
-};
-
 //----------------------------------------------------------------------------
 int vtkCGNSReader::RequestData(vtkInformation* vtkNotUsed(request),
   vtkInformationVector** vtkNotUsed(inputVector), vtkInformationVector* outputVector)
